components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,446 @@
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OAuthVersions.VERSION_2;
+import static org.wso2.carbon.identity.oauth2.util.OAuth2Util.getTenantId;
+
+/**
+ * Unit tests for AuthorizationCodeDAOImpl.
+ */
+@WithCarbonHome
+@PrepareForTest({IdentityDatabaseUtil.class, OAuth2Util.class, OAuth2TokenUtil.class, IdentityUtil.class,
+        OAuthServerConfiguration.class})
+public class AuthorizationCodeDAOImplTest extends PowerMockIdentityBaseTest {
+    public static Map<String, BasicDataSource> dataSourceMap = new HashMap<>();
+
+    @Mock
+    private ServiceProvider mockedServiceProvider;
+
+    @Mock
+    private AuthenticatedUser mockedAuthenticatedUser;
+
+    @Mock
+    private ApplicationManagementService applicationManagementService;
+
+    private Connection connection;
+
+    private AuthorizationCodeDAOImpl authorizationCodeDAO;
+    private String[] scopes;
+    AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+    private static final int DEFAULT_TENANT_ID = 1234;
+    private static final String APP_NAME = "myApp";
+    private static final String USER_NAME = "user1";
+    private static final String CALLBACK = "http://localhost:8080/redirect";
+    private static final String DB_NAME = "testAuthzCODEDB";
+
+    @BeforeClass
+    public void initTest() throws Exception {
+
+        //Initializing the database.
+        DAOUtils.initializeDataSource(DB_NAME, DAOUtils.getFilePath("identity.sql"));
+        authorizationCodeDAO = new AuthorizationCodeDAOImpl();
+        scopes = new String[]{"sms", "openid", "email"};
+        authenticatedUser.setTenantDomain("super.wso2");
+        authenticatedUser.setUserName("randomUser");
+        authenticatedUser.setUserStoreDomain("PRIMARY");
+        storeIDP();
+    }
+
+    @BeforeMethod
+    public void setup() throws Exception {
+        connection = DAOUtils.getConnection(DB_NAME);
+        mockStatic(IdentityDatabaseUtil.class);
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(IdentityDatabaseUtil.getDBConnection(false)).thenReturn(connection);
+    }
+
+    @AfterMethod
+    public void closeup() throws Exception {
+        connection.close();
+    }
+
+    @AfterClass
+    public void tearDown() throws Exception {
+        closeH2Base(DB_NAME);
+    }
+
+    public static void closeH2Base(String databaseName) throws Exception {
+        BasicDataSource dataSource = dataSourceMap.get(databaseName);
+        if (dataSource != null) {
+            dataSource.close();
+        }
+    }
+
+    private AuthzCodeDO persistAuthorizationCode(String consumerKey, String authzCodeId, String authzCode,
+                                                 boolean createApplication, String status) throws Exception {
+        if (createApplication) {
```

Comment:
why do we need this condition here? I couldn't find anywhere in this class where `boolean createApplication` will be passed as `false`.

Reply:
Addressed with bd9a919ce82d35f572e275a65b173e7fdf9e50dc